### PR TITLE
fix(payment): INT-5854 [Mollie] Klarna is not available if cart contains digital products

### PR DESCRIPTION
**Pull request by bc-alejandroRodriguez**
## What? [INT-5854](https://bigcommercecloud.atlassian.net/browse/INT-5854)
I want to hide any Klarna Pay Later and Klarna Slice It from shoppers if their cart contains digital products

## Why?
So that I can prevent shoppers from trying to pay for an ineligible cart and getting an error

## Testing / Proof

Before
https://user-images.githubusercontent.com/104527753/180102690-63e27651-3e1d-466b-bc3d-24f71aaf2b78.mov

After
https://user-images.githubusercontent.com/104527753/180099759-592b8826-6aea-4258-a53c-4835332259d4.mov

@bigcommerce/checkout @bigcommerce/apex-integrations

### CODE REVIEW

File: packages/core/src/payment/strategies/mollie/mollie-payment-strategy.spec.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ import {  MollieClient, MollieElement, MollieHostWindow } from './mollie';
 import MolliePaymentStrategy from './mollie-payment-strategy';
 import MollieScriptLoader from './mollie-script-loader';
 import { getHostedFormInitializeOptions, getInitializeOptions, getMollieClient, getOrderRequestBodyAPMs, getOrderRequestBodyVaultedCC, getOrderRequestBodyVaultAPMs, getOrderRequestBodyVaultCC, getOrderRequestBodyWithoutPayment, getOrderRequestBodyWithCreditCard } from './mollie.mock';
+import {getCart} from "../../../cart/carts.mock";
```
Comment (vmparra) [inline packages/core/src/payment/strategies/mollie/mollie-payment-strategy.spec.ts:26]: Just add spaces before and after curly braces 💅🏻

**Review by vmparra — CHANGES_REQUESTED** [review]
I loved what you did, I just left one comment.

**Review by vmparra — APPROVED** [review]
LGTM 👍🏻

File: packages/core/src/payment/strategies/mollie/mollie-payment-strategy.ts [hunk 2]
```diff
@@ -80,6 +85,32 @@ export default class MolliePaymentStrategy implements PaymentStrategy {
...
+        if(methodsNotAllowedWhenDigitalOrder.includes(methodId)) {
+            const cart = state.cart.getCart();
+
+            if (!cart) {
+                throw new MissingDataError(MissingDataErrorType.MissingCart);
+            }
+
+            const cartDigitalItems = cart.lineItems.digitalItems;
+
+            if(cartDigitalItems && cartDigitalItems.length > 0){
```
Comment (animesh1987) [inline packages/core/src/payment/strategies/mollie/mollie-payment-strategy.ts:97]:
```suggestion
            if (cartDigitalItems && cartDigitalItems.length > 0){
```

File: packages/core/src/payment/strategies/mollie/mollie-payment-strategy.ts [hunk 3]
```diff
@@ -80,6 +85,32 @@ export default class MolliePaymentStrategy implements PaymentStrategy {
             this._mountElements();
         }
 
+        if(methodsNotAllowedWhenDigitalOrder.includes(methodId)) {
```
Comment (animesh1987) [inline packages/core/src/payment/strategies/mollie/mollie-payment-strategy.ts:88]:
```suggestion
        if (methodsNotAllowedWhenDigitalOrder.includes(methodId)) {
```

File: packages/core/src/payment/strategies/mollie/mollie-payment-strategy.ts [hunk 4]
```diff
@@ -80,6 +85,32 @@ export default class MolliePaymentStrategy implements PaymentStrategy {
...
+            const cart = state.cart.getCart();
+
+            if (!cart) {
+                throw new MissingDataError(MissingDataErrorType.MissingCart);
+            }
+
```
Comment (animesh1987) [inline packages/core/src/payment/strategies/mollie/mollie-payment-strategy.ts:94]:
```suggestion
            const cart = state.cart.getCartOrThrow();
```

**Review by animesh1987 — APPROVED** [review]

**Comment by BC-ISilva** [comment]
💚 Tested!

QE Details:
User story has been validated, following scenarios have been tested:

1. If the cart consists solely of one or more digital products and the shopper selects Klarna Pay Later or Klarna Slice It, then the following message will appear within the component and the Place Order button will be grayed out.
2. If the cart is mixed with both digital and physical products and the shopper selects Klarna Pay Later or Klarna Slice It, then the following message will appear within the component and the Place Order button will be grayed out
3. If the shopper applies a coupon code in checkout which adds a digital product after Klarna Pay Later or Klarna Slice It has already appeared as an option, then the shopper will be shown an error
4. Klarna Pay Now does not look at whether there is a digital product or not in the cart.
5. Shoppers select Klarna Pay later or Slice it and then applies a coupon to add digital items as gif.

Please review attached word document with validation details:
[INT-5854 [Mollie] Klarna Is Not Available If Cart Contains Digital Products.docx](https://github.com/bigcommerce/checkout-sdk-js/files/9340088/INT-5854.Mollie.Klarna.Is.Not.Available.If.Cart.Contains.Digital.Products.docx)

**Pull request merged**